### PR TITLE
WebKit export of https://bugs.webkit.org/show_bug.cgi?id=181067

### PR DESCRIPTION
**Pull request by youennf**
Service Worker should expose redirect mode for navigation loads as manual

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/22049)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/319472268)

**Comment by youennf** [comment]
Closed in favor of https://github.com/w3c/web-platform-tests/pull/9332

File: service-workers/service-worker/request-end-to-end.https.html [hunk 1]
```diff
@@ -23,13 +23,12 @@
...
-          assert_equals(result.request_construct_error, 'TypeError',
+          assert_equals(result.request_construct_error, '',
                         'Constructing a Request with a Request whose mode ' +
-                        'is navigate and non-empty RequestInit must throw a ' +
-                        'TypeError.')
+                        'is navigate and non-empty RequestInit should not throw');
           assert_equals(result.credentials, 'include', 'request.credentials');
           assert_equals(result.redirect, 'manual', 'request.redirect');
-          assert_equals(result.headers['user-agent'], undefined,
+          assert_true(result.headers['user-agent'] === undefined,
```
Comment (wanderview) [inline service-workers/service-worker/request-end-to-end.https.html]: @youennf Why this change from `assert_equals(t, undefined)` to `assert_true(t === undefined)`?
